packages/backend-common/src/database/postgres.ts

```diff
@@ -54,10 +69,35 @@ export function buildPgDatabaseConfig(
 }
 
 /**
- * Gets the postgres connection config
+ * Builds a knex Postgres database connection for database creation
+ *
+ * @param dbConfig The database config
+ */
+function buildPgDatabaseAdminConfig(dbConfig: JsonValue) {
+  return mergeDatabaseConfig(dbConfig, {
+    connection: {
+      database: 'postgres',
```

Comment:
I'm not sure it's safe to assume the default db would be `postgres`. I'd personally feel better if this was configurable. I think introducing a `backend.adminDatabase` config setting might be helpful in this regard.

Reply:
When I did my quick hack I didn't specify a database at all (maybe knex defaulted to something). The host, port and credentials was enough to be able to query the `pg_database` catalog to check if the databases exists.

Reply:
I believe you can connect just to the postgres server and list/change dbs afterwards so that's probably what happened here. Yeah, I'm not sure it's required, but also wouldn't assume the standard `postgres` setup will always be used. For instance AWS RDS does some things a little differently then if you say had it self hosted using stock pg.

Reply:
I can do that; though I have never heard of the admin database being other than postgres, even in RDS and a quick search doesn't yield anything meaningful.

Reply:
Sorry, the db is probably not a great examples, but there are some differences regarding role defaults etc. I think the point is why not just let the user supply the db via an `adminDatabase` config that is structured exactly like the standard `backend.database` config? That way most of this can just leverage the existing functions and not require admin specific logic.

Reply:
Hmm. In this new design wouldn't the `backend.database` be pointing to an admin config (credentials, etc?), since the plugins are now using a separate role? What would the additional `backend.adminDatabase` config be for?

Reply:
Hmm, I may have overlooked the roles here. But the way I'm reading it, the new role login would have the same password? I would think having completely different admin / app credentials would be ideal. Possibly to the point that each plugin db has it's own (but seems better left for a future effort).

Reply:
So the flow now looks like this:

 - app-config.yaml: contains "root" permissions (typically the "postgres" user)
 - createPgDatabaseClient: accepts "root" permissions, generates "plugin" role and "plugin" database if they do not exist. returns knex using "plugin" role/password.

So plugins do not have a change in logic; instead of running with root permissions before, they've been downgraded to a plugin-specific user.